cli/src/commands/mod.rs

```diff
@@ -3190,10 +3190,10 @@ fn cmd_restore(
             .rewrite_commit(command.settings(), &to_commit)
             .set_tree_id(new_tree_id)
             .write()?;
+        tx.finish(ui)?;
         write!(ui.stderr(), "Created ")?;
-        tx.write_commit_summary(ui.stderr_formatter().as_mut(), &new_commit)?;
+        workspace_command.write_commit_summary(ui.stderr_formatter().as_mut(), &new_commit)?;
```

Comment:
Another workaround is to call `rebase_descendants()` manually (we can add `tx.rebase_descendants()` as needed.) I think the message order is slightly better than emitting "Created ..." after checkout.

Reply:
Thanks for the tip, will try it! I agree that the old order is preferable if we can make it work.

Reply:
Yes, it worked, causing a few more curious changes (seemingly for the better).